src/Microsoft.ML.Data/DataView/CompositeRowToRowMapper.cs

```diff
@@ -49,18 +49,19 @@ IEnumerable<DataViewSchema.Column> IRowToRowMapper.GetDependencies(IEnumerable<D
             return columnsNeeded;
         }
 
-        public DataViewRow GetRow(DataViewRow input, Func<int, bool> active)
+        DataViewRow IRowToRowMapper.GetRow(DataViewRow input, IEnumerable<DataViewSchema.Column> activeColumns)
```

Comment:
>IEnumerable<DataViewSchema.Column> activeColumns [](start = 62, length = 48)

It's probably quite late considering how much work you did already for this issue.
So we replace functions to schema columns.
In past we had just indexes and we just check are they belong to schema range or not.
Now we accept schema columns. But we use only index position in them.
Which can lead to situation where i have two dataviews with following schemas: 
one is `A: int, B: float`,  and another is `C:string, D: bool`.
I want to get rows from first dataview. but I made mistake and I pass schema from second dataview as active columns.
I would assume it would work perfectly fine in current state of the code. I'm not sure it should.

Reply:
Agree. should switch from tracking activeIndices to active columns at some point. I'll log an issue. 

---
In reply to: [261487085](https://github.com/dotnet/machinelearning/pull/2796#discussion_r261487085) [](ancestors = 261487085)